databricks/koalas/frame.py

```diff
@@ -9328,34 +9328,58 @@ def filter(self, items=None, like=None, regex=None, axis=None):
             else:
                 raise ValueError("items should be a list-like object.")
             if axis == 0:
-                # TODO: support multi-index here
-                if len(index_scols) != 1:
-                    raise ValueError("Single index must be specified.")
-                col = None
-                for item in items:
-                    if col is None:
-                        col = index_scols[0] == F.lit(item)
-                    else:
-                        col = col | (index_scols[0] == F.lit(item))
+                if len(index_scols) == 1:
+                    col = None
+                    for item in items:
+                        if col is None:
+                            col = index_scols[0] == F.lit(item)
+                        else:
+                            col = col | (index_scols[0] == F.lit(item))
+                elif len(index_scols) > 1:
+                    # for multi-index
+                    cols = []
+                    col = None
+                    for item in items:
+                        if col is None:
+                            for i, element in enumerate(item):
+                                if col is None:
```

Comment:
Should we check this here again? I think this is already checked at line 9343.

Reply:
I've  just removed double check